web/gui-v2/src/components/ListViewTable.jsx

```diff
@@ -0,0 +1,388 @@
+      stage: listToDropdownOptions(stages),
+    }),
+    [continents, stages]
+  );
+  // console.info("filterOptions:", filterOptions);
+
+
+  // Prepare the columns that we will display in the `<Table>`, including
+  // adding the appropriate filter mechanisms to the header cells.
+  const columns = columnDefinitions
+    .filter(colDef => columnsParam.includes(colDef.key))
+    .map((colDef) => {
+      let display_name;
+      switch ( colDef.type ) {
+        case 'dropdown':
+          display_name = (
+            <HeaderDropdown
+              label={colDef.title}
+              options={filterOptions?.[colDef.key]}
+              selected={filters?.[colDef.key].get}
+              setSelected={newVal => handleDropdownChange(colDef.key, newVal)}
+            />
+          );
+          break;
+        case 'slider':
+          display_name = (
+            <HeaderSlider
+              label={colDef.title}
+              onChange={newVal => handleSliderChange(colDef.key, newVal)}
+              value={filters?.[colDef.key].get}
+            />
+          );
+          break;
+        case 'stock':
+          display_name = (
+            colDef.title
+          );
+          break;
+        default:
+          display_name = colDef.title;
+      }
+      const column = {
+        display_name,
+        key: colDef.key,
+        sortable: colDef.sortable,
+        css: colDef.type === 'slider' && css`width: 120px;`,
+      };
+      if ( colDef?.format ) {
+        column.format = colDef.format;
+      } else if ( colDef.key === "name" ) {
+        column.format = (name) => <Link to="company/163">{name}</Link>
+      }
+      return column;
+    });
+
+  // Prepare a mapping of column key --> column index; necessary for the table
+  // sort comparator to be able to extract the correct values from each row.
+  const colDefIndices = {};
+  columnDefinitions.forEach((def, ix) => {
+    colDefIndices[def.key] = ix;
+  });
+
+
+  // Some columns, such as `ai_pubs`, contain an object with multiple subvalues,
+  // which cannot be sorted with the default comparator.  Define a column-aware
+  // comparator that will let us sort entries correctly.
+  const tableSortComparator = (a, b, key) => {
+    const colDef = columnDefinitions[colDefIndices[key]];
+    const aVal = colDef?.extract ? colDef.extract(a[key]) : a[key];
+    const bVal = colDef?.extract ? colDef.extract(b[key]) : b[key];
+
+    const aIsUndef = aVal === null || aVal === undefined;
+    const bIsUndef = bVal === null || bVal === undefined;
+
+    if ( aIsUndef && bIsUndef ) {
+      return 0;
+    } else if ( aIsUndef ) {
+      return 1;
+    } else if ( bIsUndef ) {
+      return -1;
+    } else if ( bVal < aVal ) {
+      return -1;
+    } else if ( aVal < bVal ) {
+      return 1;
+    } else {
+      return 0;
+    }
+  };
+
+  const resetFilters = () => {
+    columnDefinitions.forEach((colDef) => {
+      if ( colDef.type === "dropdown" ) {
```

Comment:
Maybe you could use an object to specify default values both here and where `filters` is defined on 115 to make sure these stay in sync?

Reply:
Tracked in https://github.com/georgetown-cset/parat/issues/26